driver-async/src/main/com/mongodb/async/client/MongoClients.java

```diff
@@ -155,6 +156,8 @@ public static MongoClient create(final ConnectionString connectionString, final
         if (connectionString.getStreamType() != null) {
             if (connectionString.getStreamType().toLowerCase().equals("netty")) {
                 builder.streamFactoryFactory(NettyStreamFactoryFactory.builder().build());
+            } else if (connectionString.getStreamType().toLowerCase().equals("nio2")) {
+                builder.streamFactoryFactory(new AsynchronousSocketChannelStreamFactoryFactory());
```

Comment:
Can we test this in MongoClientsSpecification by temporarily setting the 'org.mongodb.async.type' system property to 'netty', and then assert that the default stream factory factory in MongoClientSettings.Builder is changed back to an instance of AsynchronousSocketChannelStreamFactoryFactory()?